zig: use @splat more

matklad opened the pull request.

matklad committed:
multiversion: unit-test!

matklad committed:
multiversion: simplify a bit by removing bounded arrays

matklad commented:
Got carried away a bit on stream, and also threw in removal of bounded arrays, which are one of my lest favorite type in zig's std :-]

matklad commented on src/unit_tests.zig:
```diff
@@ -13,6 +13,7 @@ comptime {
     _ = @import("time.zig");
     _ = @import("git-review.zig");
     _ = @import("message_buffer.zig");
+    _ = @import("multiversioning.zig");
```
Ouch!

batiati commented:
Humm, I missed this pattern `std.mem.zeroes([N]foo)` when I did the `@splat` replace.
If you grep for `std.mem.zeroes`, you might find many more instances that could use `@splat`.

matklad committed:
zig: avoid mem.zeros in favor of @splat

It's a smelly function, and usually it's better to be explicit about
what you are zeroing out. It's not bad enough to earn a tidy ban, but
these splats are a clear improvement!

matklad commented:
Good call, Rafael!

cb22 commented on src/multiversioning.zig:
```diff
@@ -204,65 +198,54 @@ pub const MultiversionHeader = extern struct {
+        git_commits: [past_releases_max][20]u8 = @splat(@splat(0)),
+        release_client_mins: [past_releases_max]u32 = @splat(0),
+
+        pub fn add(self: *PastReleases, next: struct {
```
Wild how obvious this seems in retrospect!

cb22 commented:
LGTM - needs a rebase.

matklad committed:
test: sort

matklad commented:
rebased!
